### PR TITLE
Change the week header and left button style to meet the date spacing

### LINKED ISSUE #27536: Incorrect Calendar Month Grid position
Calendar Month Grid has an incorrect position in RTL, WprdPress 5.5.3, the problem shot:
image

It came from incorrect value of `transform` for `CalendarMonthGrid CalendarMonthGrid_1 CalendarMonthGrid__horizontal CalendarMonthGrid__horizontal_2`.

After edit the transform value:
image

Update:
You can see this problem in LTR too:
image

### PR DESCRIPTION
**Pull request by hsingyuc**
## Description
Fix #27536 
Change the week header and left button style to meet the date spacing.

## How has this been tested?
Tested in English and in Arabic (rtl), both render as expected

## Screenshots 
1. Before
before_change
1. After
change_spacing

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Review by jasmussen — COMMENTED** [review]
This is awesome! Small, focused PR! Thank you. Before:

before

After:

after

_The month navigation buttons have better alignment_.

I'm seeing an issue in arabic, though:

Screenshot 2021-01-05 at 08 52 00

The button to navigate _left_ takes up the whole screen. This may have been a pre-existing issue, but because this PR aims to close the ticket that reported the issue, it seems worth fixing.

From a little debugging in the inspector, I found this rule:

```
.components-datetime__date .DayPickerNavigation_leftButton__horizontalDefault {
    right: 13px;
}
```

If that gets changed to:

```
.components-datetime__date .DayPickerNavigation_leftButton__horizontalDefault {
    left: 13px;
}
```

all is fine. 

In other words, it appears that the `left` and `right` positioning gets auto-rewritten for RTL, when it shouldn't be. So in the code where it says `left`, you can add the following line before it, to have it not be rewritten: `/*rtl:ignore*/`.

If we get that in, we can ship this. Thanks again.

**Comment by hsingyuc** [comment]
@jasmussen  Thanks for the tip! The change has been pushed.

**Comment by jasmussen** [comment]
LTR is good:

ltr

But I'm still seeing an issue with RTL:

rtl

It looks like you need to not only be explicit about the left value, but the right value as well. 

This might work:

```
		/*!rtl:begin:ignore*/
		left: 13px;
		right: auto;
		/*!rtl:end:ignore*/
```

Thanks for your patience!

**Comment by hsingyuc** [comment]
@jasmussen  Thank you! The begin/end ignore is a good addition! I've pushed that change. Could you confirm if you have pulled the latest changes and built the scripts?  I see the arrows properly aligned in RTL without the need for `right: auto;`. Thank you for your time!

**Comment by jasmussen** [comment]
Thanks again. I'm about to head out for the evening but promise I will test properly tomorrow morning my time, and approve if it works as I think it will. But in the mean time, someone else should feel free to test and approve.

**Comment by hsingyuc** [comment]
It looks like this PR may also resolve https://github.com/WordPress/gutenberg/issues/27535.

Also, I ran into another problem with the RTL calendar.  The current month appears hidden because of the positioning of months in RTL mode.

螢幕快照 2021-01-06 下午3 18 46

Do you see this issue?

I think there's a quick fix for this by absolutely positioning the third calendar month in the container since we don't use calendar animations in GB, but this problem may be upstream in react dates.  https://github.com/airbnb/react-dates/pull/1552.

**Review by jasmussen — APPROVED** [review]
Thanks so much for this PR, and thanks for the patience. With the latest changes, this is solid in LTR:

Screenshot 2021-01-07 at 09 10 04

And RTL:

Screenshot 2021-01-07 at 09 10 31

Here's RTL before, for comparison:

Screenshot 2021-01-07 at 09 24 58

That arabic calendar does look fairly compressed to me, though, so there may be a separeate bug at play, but since that existed before this PR it's unrelated. Admittedly my Arabic is a little rusty, so I'll defer to speakers of that language.

Ship this, and thanks again!

**Comment by hsingyuc** [comment]
The pleasure is all mine 🤝
